### PR TITLE
BL-1132 test and fix for Bogus Bubble Bug

### PR DESCRIPTION
**Pull request by gmartin7**

### CODE REVIEW

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/348/
Test FAILed.

**Comment by JohnThomson** [comment]
retest this please

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/349/
Test PASSed.

**Comment by gmartin7** [comment]
That's better!

File: src/BloomExe/Book/TranslationGroupManager.cs [hunk 1]
```diff
@@ -244,16 +244,32 @@ private static void MakeElementWithLanguageForOneGroup(XmlElement groupElement,
...
+			foreach (var node in removalList)
 			{
-				node.ParentNode.RemoveChild(node);
+				try
+				{
+					node.ParentNode.RemoveChild(node);
+				}
+				catch (ArgumentException)
+				{
+					// Means we already deleted an intermediate node containing this one
```
Comment (JohnThomson) [inline src/BloomExe/Book/TranslationGroupManager.cs]: Doesn't seem possible for it to mean that now we're using ParentNode to do the removal.

File: src/BloomExe/Book/TranslationGroupManager.cs [hunk 2]
```diff
@@ -244,16 +244,32 @@ private static void MakeElementWithLanguageForOneGroup(XmlElement groupElement,
...
+		}
+
+		private static void RemoveXmlChildren(List<XmlNode> removalList)
+		{
+			foreach (var node in removalList)
 			{
-				node.ParentNode.RemoveChild(node);
+				try
+				{
+					node.ParentNode.RemoveChild(node);
```
Comment (JohnThomson) [inline src/BloomExe/Book/TranslationGroupManager.cs]: Is there any scenario of structure and order of deletions in which ParentNode could be null?

Comment (gmartin7) [inline src/BloomExe/Book/TranslationGroupManager.cs]: I put in a test for null ParentNode. I don't know of a case where it'd be necessary, but always good to be safe.

File: src/BloomExe/Book/TranslationGroupManager.cs (hunk 1 above)
Comment (gmartin7) [inline src/BloomExe/Book/TranslationGroupManager.cs]: Removed the try-catch. You're right, it shouldn't be possible to cause that exception now.

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/356/
Test PASSed.

File: src/BloomExe/Book/TranslationGroupManager.cs [hunk 3]
```diff
@@ -244,16 +244,25 @@ private static void MakeElementWithLanguageForOneGroup(XmlElement groupElement,
...
+			RemoveXmlChildren(listToRemove);
+		}
+
+		private static void RemoveXmlChildren(List<XmlNode> removalList)
+		{
+			foreach (var node in removalList)
 			{
-				node.ParentNode.RemoveChild(node);
+				if(node.ParentNode != null)
+					node.ParentNode.RemoveChild(node);
```
Comment (StephenMcConnel) [inline src/BloomExe/Book/TranslationGroupManager.cs]: I would prefer a space between the if and the (, but that's not enough to hold up committing this change.

**Comment by StephenMcConnel** [comment]
looks good to me.

File: src/BloomExe/Book/TranslationGroupManager.cs (hunk 3 above)
Comment (gmartin7) [inline src/BloomExe/Book/TranslationGroupManager.cs]: Me too! Not sure how that slipped past me. That's usually part of my style. Maybe I'll fix it later...

**Pull request merged**